### PR TITLE
Add human-readable format helpers.

### LINKED ISSUE #3: BinaryByteUnit.format(long count)
I want something that passes this test:

```
assertThat(BinaryByteUnit.format(16384)).isEqualTo("16 KiB");
assertThat(BinaryByteUnit.format(1288490189)).isEqualTo("1.2 GiB");
```

Maybe also this:

```
assertThat(BinaryByteUnit.format(1288490189, "#.##")).isEqualTo("1.20 GiB");
```

### PR DESCRIPTION
**Pull request by JakeWharton**
Closes #3. @swankjesse

### CODE REVIEW

File: src/main/java/com/jakewharton/byteunits/BitUnit.java [hunk 1]
```diff
@@ -312,4 +314,36 @@ public long toTerabits(long count) {
...
+    return format(bytes, new DecimalFormat(pattern));
+  }
+
+  /**
+   * Return {@code bytes} as human-readable size string (e.g., "1.2 Gb". This will use {@code
+   * format} for formatting the number.
+   */
+  public static String format(long bytes, DecimalFormat format) {
+    int unitIndex = 0;
+    double count = bytes;
```
Comment (swankjesse) [inline src/main/java/com/jakewharton/byteunits/BitUnit.java]: divide by 8 ?

Comment (swankjesse) [inline src/main/java/com/jakewharton/byteunits/BitUnit.java]: or multiply by 8?

Comment (swankjesse) [inline src/main/java/com/jakewharton/byteunits/BitUnit.java]: (bits are hard to reason about)

Comment (swankjesse) [inline src/main/java/com/jakewharton/byteunits/BitUnit.java]: ... or maybe the parameter should be `bits` instead of `bytes` ?!

File: src/test/java/com/jakewharton/byteunits/BinaryByteUnitTest.java [hunk 2]
```diff
@@ -44,4 +47,30 @@
...
   }
+
+  @Test public void format() {
+    assertEquals("0 B", BinaryByteUnit.format(0));
+    assertEquals("1 B", BinaryByteUnit.format(1));
+    assertEquals("1 KiB", BinaryByteUnit.format(1024));
+    assertEquals("1.001 KiB", BinaryByteUnit.format(1025));
+    assertEquals("16 KiB", BinaryByteUnit.format(16_384));
+    assertEquals("1.177 MiB", BinaryByteUnit.format(1_234_567));
+    assertEquals("1.2 GiB", BinaryByteUnit.format(1_288_490_189));
```
Comment (swankjesse) [inline src/test/java/com/jakewharton/byteunits/BinaryByteUnitTest.java]: dem underscores

File: src/test/java/com/jakewharton/byteunits/BinaryByteUnitTest.java [hunk 3]
```diff
@@ -44,4 +47,30 @@
...
+    assertEquals("1.0 KiB", BinaryByteUnit.format(1024, pattern));
+    assertEquals("1.0 KiB", BinaryByteUnit.format(1025, pattern));
+    assertEquals("16.0 KiB", BinaryByteUnit.format(16_384, pattern));
+    assertEquals("1.18 MiB", BinaryByteUnit.format(1_234_567, pattern));
+  }
+
+  @Test public void formatWithDecimalFormat() {
+    DecimalFormat format = new DecimalFormat("#.##", new DecimalFormatSymbols(Locale.FRENCH));
+    assertEquals("16 KiB", BinaryByteUnit.format(16_384, format));
+    assertEquals("1,18 MiB", BinaryByteUnit.format(1_234_567, format));
```
Comment (swankjesse) [inline src/test/java/com/jakewharton/byteunits/BinaryByteUnitTest.java]: c’est bon

File: src/test/java/com/jakewharton/byteunits/BitUnitTest.java [hunk 4]
```diff
@@ -50,4 +53,30 @@
...
   }
+
+  @Test public void format() {
+    assertEquals("0 b", BitUnit.format(0));
+    assertEquals("1 b", BitUnit.format(1));
+    assertEquals("1 Kb", BitUnit.format(1000));
+    assertEquals("1.001 Kb", BitUnit.format(1001));
+    assertEquals("16 Kb", BitUnit.format(16_000));
+    assertEquals("1.177 Mb", BitUnit.format(1_177_171));
+    assertEquals("1.2 Gb", BitUnit.format(1_200_000_000));
```
Comment (swankjesse) [inline src/test/java/com/jakewharton/byteunits/BitUnitTest.java]: Do Long.MAX_VALUE to test the boundary cases?

File: src/main/java/com/jakewharton/byteunits/BinaryByteUnit.java [hunk 5]
```diff
@@ -272,4 +274,36 @@ public long toTebibytes(long count) {
...
+   */
+  public static String format(long bytes, String pattern) {
+    return format(bytes, new DecimalFormat(pattern));
+  }
+
+  /**
+   * Return {@code bytes} as human-readable size string (e.g., "1.2 GiB". This will use {@code
+   * format} for formatting the number.
+   */
+  public static String format(long bytes, DecimalFormat format) {
```
Comment (swankjesse) [inline src/main/java/com/jakewharton/byteunits/BinaryByteUnit.java]: what should happen if bytes is negative?

**Comment by swankjesse** [comment]
LGTM

File: src/main/java/com/jakewharton/byteunits/BinaryByteUnit.java (hunk 5 above)
Comment (JakeWharton) [inline src/main/java/com/jakewharton/byteunits/BinaryByteUnit.java]: IAE

File: src/test/java/com/jakewharton/byteunits/BitUnitTest.java (hunk 4 above)
Comment (JakeWharton) [inline src/test/java/com/jakewharton/byteunits/BitUnitTest.java]: Done

File: src/main/java/com/jakewharton/byteunits/BitUnit.java (hunk 1 above)
Comment (JakeWharton) [inline src/main/java/com/jakewharton/byteunits/BitUnit.java]: Yep, just should be bits.

**Comment by JakeWharton** [comment]
I changed the default format to only use a single decimal digit.

File: src/test/java/com/jakewharton/byteunits/BinaryByteUnitTest.java [hunk 6]
```diff
@@ -44,4 +49,53 @@
...
+
+  @Test public void format() {
+    assertEquals("0 B", BinaryByteUnit.format(0));
+    assertEquals("1 B", BinaryByteUnit.format(1));
+    assertEquals("1 KiB", BinaryByteUnit.format(1024));
+    assertEquals("1 KiB", BinaryByteUnit.format(1025));
+    assertEquals("16 KiB", BinaryByteUnit.format(16_384));
+    assertEquals("1.2 MiB", BinaryByteUnit.format(1_234_567));
+    assertEquals("1.2 GiB", BinaryByteUnit.format(1_288_490_189));
+    assertEquals("8,192 PiB", BinaryByteUnit.format(Long.MAX_VALUE));
```
Comment (swankjesse) [inline src/test/java/com/jakewharton/byteunits/BinaryByteUnitTest.java]: sweeeet.

File: src/test/java/com/jakewharton/byteunits/DecimalByteUnitTest.java [hunk 7]
```diff
@@ -44,4 +49,53 @@
...
+
+  @Test public void format() {
+    assertEquals("0 B", DecimalByteUnit.format(0));
+    assertEquals("1 B", DecimalByteUnit.format(1));
+    assertEquals("1 KB", DecimalByteUnit.format(1000));
+    assertEquals("1 KB", DecimalByteUnit.format(1001));
+    assertEquals("16 KB", DecimalByteUnit.format(16_000));
+    assertEquals("1.2 MB", DecimalByteUnit.format(1_177_171));
+    assertEquals("1.2 GB", DecimalByteUnit.format(1_200_000_000));
+    assertEquals("9,223.4 PB", DecimalByteUnit.format(Long.MAX_VALUE));
```
Comment (swankjesse) [inline src/test/java/com/jakewharton/byteunits/DecimalByteUnitTest.java]: off topic: compounded, the extra 0.024% makes quite a difference!

**Comment by swankjesse** [comment]
Thanks!

**Comment by JakeWharton** [comment]
Will do a release when I'm not on a plane.
